scripts/release.sh

```diff
@@ -136,6 +136,41 @@ function setup_branch() {
   git fetch ${KNATIVE_UPSTREAM} ${RELEASE_BRANCH}:upstream/${RELEASE_BRANCH}
 }
 
+# Setup version, branch and release notes for a auto release.
+function prepare_auto_release() {
+  echo "Auto release requested"
+  TAG_RELEASE=1
+  PUBLISH_RELEASE=1
+
+  local tags="$(git tag | cut -d 'v' -f2 | cut -d '.' -f1-2 | sort | uniq )"
+  local branches="$( { (git branch -r | grep origin/release-) ; (git branch  | grep release-); } | cut -d '-' -f2 | sort | uniq)"
+  local last_version=""
+
+  for i in $branches; do
+    RELEASE_VERSION=$i
+    for j in $tags; do
+      last_version="v${j}.0"
+      if [[ "$i" == "$j" ]]; then
+        RELEASE_VERSION=""
+      fi
+    done
+  done
+
+  if [ "$RELEASE_VERSION" == "" ];then
+    echo "Everything is released. Nothing to do!"
+    exit  0
+  fi
+
+  RELEASE_BRANCH="release-${RELEASE_VERSION}"
+  echo "Will create release ${RELEASE_VERSION} from branch ${RELEASE_BRANCH}"
+  # If --release-notes not used, copy from the latest release
```

Comment:
Better not. Things usually change a log between major releases, this will just make things harder to edit. Use a placeholder markdown file instead.